src/utils/input_parser.cpp

```diff
@@ -580,13 +580,13 @@ void parse(Input& input, const std::string& input_str, bool geometry) {
         }
       }
     }
-  } else {
+  } else if (json_input.HasMember("matrix")) {
     // Deprecated `matrix` key still interpreted as
     // `matrices.DEFAULT_PROFILE.duration` for retro-compatibility.
-    if (json_input.HasMember("matrix")) {
-      input.set_durations_matrix(DEFAULT_PROFILE,
-                                 get_matrix<Duration>(json_input["matrix"]));
-    }
+    input.set_durations_matrix(DEFAULT_PROFILE,
+                                get_matrix<Duration>(json_input["matrix"]));
+  } else {
```

Comment:
Not really a fan of a logic branch that does nothing, and I don't think we really need to comment here on stuff that other parts of the code is responsible for. We can probably revert all changes to `input_parser.cpp`?